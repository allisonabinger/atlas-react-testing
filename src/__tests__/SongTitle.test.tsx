// Test Suite for the SongTitle Component

import { render } from "@testing-library/react";
import { expect, test, describe } from "vitest";
import SongTitle from "../components/SongTitle";

// Snapshot Tests for CoverArt
describe("SongTitle Component", () => {
  test("SongTitle renders the title and artist", () => {
    const { container } = render(
      <SongTitle title="TestSong" artist="TestArtist" />,
    );
    expect(container).toMatchSnapshot();
  });

  test("CoverArt will not render without a valid title and artist", () => {
    const { container } = render(<SongTitle title="" artist="" />);
    expect(container).toMatchSnapshot();
  });

  test("CoverArt will not render without a valid title", () => {
    const { container } = render(<SongTitle title="" artist="TestArtist" />);
    expect(container).toMatchSnapshot();
  });

  test("CoverArt will not render without a valid artist", () => {
    const { container } = render(<SongTitle title="TestSong" artist="" />);
    expect(container).toMatchSnapshot();
  });
});
