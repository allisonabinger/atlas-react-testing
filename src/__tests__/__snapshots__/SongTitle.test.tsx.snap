// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SongTitle Component > CoverArt will not render without a valid artist 1`] = `
<div>
  <div>
    <h2
      class="leading-8 text-2xl font-bold mb-2 text-vista-blue-900"
    >
      TestSong
    </h2>
    <p
      class="mb-4 text-vista-blue-900"
    />
  </div>
</div>
`;

exports[`SongTitle Component > CoverArt will not render without a valid title 1`] = `
<div>
  <div>
    <h2
      class="leading-8 text-2xl font-bold mb-2 text-vista-blue-900"
    />
    <p
      class="mb-4 text-vista-blue-900"
    >
      TestArtist
    </p>
  </div>
</div>
`;

exports[`SongTitle Component > CoverArt will not render without a valid title and artist 1`] = `
<div>
  <div>
    <h2
      class="leading-8 text-2xl font-bold mb-2 text-vista-blue-900"
    />
    <p
      class="mb-4 text-vista-blue-900"
    />
  </div>
</div>
`;

exports[`SongTitle Component > SongTitle renders the title and artist 1`] = `
<div>
  <div>
    <h2
      class="leading-8 text-2xl font-bold mb-2 text-vista-blue-900"
    >
      TestSong
    </h2>
    <p
      class="mb-4 text-vista-blue-900"
    >
      TestArtist
    </p>
  </div>
</div>
`;
